City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2488: Changes to service page blocks.

# [AU-2488](https://helsinkisolutionoffice.atlassian.net/browse/AU-2488)

## What was done

This pull request implements logic changes to the `ServicePageAuthBlock` and `ServicePageAnonBlock` blocks. In addition to this, the code has been cleaned up by extracting lots of duplicate code to its own service, the `ServicePageBlockService`. The blocks should now work with the following logic:

- If a service page has not referenced a Webform, then neither of the blocks are displayed.

<img width="1518" alt="Screenshot 2024-05-17 at 14 26 50" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/6dac4791-71b7-4428-87ab-dc3d43f117fc">

- If a service page does reference a Webform, but the application is not open, then a "This application is not open" message should be displayed to the user.

<img width="1482" alt="Screenshot 2024-05-17 at 14 26 17" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/018d9314-a588-4c8a-8b09-d0a506fc7c18">

- If a service page does reference a Webform and the application is open, but the user has the wrong role, then a "Change your role" message should be displayed to the user.

<img width="1496" alt="Screenshot 2024-05-17 at 14 25 46" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/6f582b90-548a-47e0-841e-e807bd138833">

- In other cases, a link to the Webform should be displayed.

<img width="1485" alt="Screenshot 2024-05-17 at 14 27 32" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/621f3e28-7e31-47e5-8ceb-4c7e6aac7322">


## How to install

Make sure your instance is up and running on correct branch.
* `feature/AU-2488-service-page-blocks`
* `make fresh`
* `make drush-cr`

## How to test

- Make sure the blocks work according to the description above. You can for example edit the [Liikunnan laitosavustus](https://hel-fi-drupal-grant-applications.docker.so/fi/tietoa-avustuksista/liikunnan-avustukset/liikunnan-laitosavustus) node and play around with the settings of the corresponding [Webform](https://hel-fi-drupal-grant-applications.docker.so/fi/admin/structure/webform/manage/liikunta_laitosavustushakemus/settings).

- Review the code.


[AU-2488]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
